### PR TITLE
Codeql v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to redmine ticket:
None

Updated lgtm.yml and codeql.yml so :
-Suricata will works on LGTM
-Codeql workflow for C will successfully run

Fixes one dangerous pattern found by CodeQL
Replaces #7183 
Goal is to fix all 18 warnings

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 6939

File: src/output-json.c [hunk 1]
```diff
@@ -173,12 +165,8 @@ void EveFileInfo(JsonBuilder *jb, const File *ff, const bool stored)
     }
 
     if (ff->flags & FILE_SHA256) {
-        size_t x;
-        int i;
         char str[256];
-        for (i = 0, x = 0; x < sizeof(ff->sha256); x++) {
-            i += snprintf(&str[i], 255-i, "%02x", ff->sha256[x]);
-        }
+        PrintHexString(str, 256, (uint8_t *)ff->sha256, sizeof(ff->sha256));
```
Comment (victorjulien) [inline src/output-json.c:169]: `sizeof(str)` instead of `256`.

File: src/util-pidfile.c [hunk 2]
```diff
@@ -104,17 +104,18 @@ void SCPidfileRemove(const char *pid_filename)
  */
 int SCPidfileTestRunning(const char *pid_filename)
 {
-    if (access(pid_filename, F_OK) == 0) {
```
Comment (victorjulien) [inline src/util-pidfile.c:107]: indent/formatting is now off I think

**Comment by victorjulien** [comment]
Build failure for me
```
detect-detection-filter.c:114:26: error: declaration shadows a local variable [-Werror,-Wshadow]
    for (size_t pos = 0, df_opt = strtok_r(copy_str, ",", &saveptr);
                         ^
detect-detection-filter.c:103:29: note: previous declaration is here
    char *copy_str = NULL, *df_opt = NULL;
                            ^
detect-detection-filter.c:114:26: error: incompatible pointer to integer conversion initializing 'size_t' (aka 'unsigned long') with an expression of type 'char *' [-Werror,-Wint-conversion]
    for (size_t pos = 0, df_opt = strtok_r(copy_str, ",", &saveptr);
                         ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
detect-detection-filter.c:115:46: error: comparison between pointer and integer ('size_t' (aka 'unsigned long') and 'void *') [-Werror,-Wpointer-integer-compare]
            pos < strlen(copy_str) && df_opt != NULL;
                                      ~~~~~~ ^  ~~~~
detect-detection-filter.c:116:27: error: incompatible pointer to integer conversion assigning to 'size_t' (aka 'unsigned long') from 'char *' [-Werror,-Wint-conversion]
            pos++, df_opt = strtok_r(NULL, ",", &saveptr)) {
                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
detect-detection-filter.c:117:19: error: incompatible integer to pointer conversion passing 'size_t' (aka 'unsigned long') to parameter of type 'const char *' [-Werror,-Wint-conversion]
        if(strstr(df_opt,"count"))
                  ^~~~~~
/usr/include/string.h:330:34: note: passing argument to parameter '__haystack' here
extern char *strstr (const char *__haystack, const char *__needle)
                                 ^
detect-detection-filter.c:119:19: error: incompatible integer to pointer conversion passing 'size_t' (aka 'unsigned long') to parameter of type 'const char *' [-Werror,-Wint-conversion]
        if(strstr(df_opt,"second"))
                  ^~~~~~
/usr/include/string.h:330:34: note: passing argument to parameter '__haystack' here
extern char *strstr (const char *__haystack, const char *__needle)
                                 ^
detect-detection-filter.c:121:19: error: incompatible integer to pointer conversion passing 'size_t' (aka 'unsigned long') to parameter of type 'const char *' [-Werror,-Wint-conversion]
        if(strstr(df_opt,"track"))
                  ^~~~~~
/usr/include/string.h:330:34: note: passing argument to parameter '__haystack' here
extern char *strstr (const char *__haystack, const char *__needle)
                                 ^
detect-detection-filter.c:103:29: error: unused variable 'df_opt' [-Werror,-Wunused-variable]
    char *copy_str = NULL, *df_opt = NULL;
                            ^
  CC       detect-dnp3.o
8 errors generated.
```

File: src/output-json.c (hunk 1 above)
Comment (catenacyber) [reply src/output-json.c:169]: ok

File: src/util-pidfile.c (hunk 2 above)
Comment (catenacyber) [reply src/util-pidfile.c:107]: fixing

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7299

**Pull request closed without merging**